mllib/src/test/scala/org/apache/spark/ml/feature/VectorAssemblerSuite.scala

```diff
@@ -261,4 +261,14 @@ class VectorAssemblerSuite
     val output = vectorAssembler.transform(dfWithNullsAndNaNs)
     assert(output.select("a").limit(1).collect().head == Row(Vectors.sparse(0, Seq.empty)))
   }
+  test("SPARK-31671: should give explicit error message when can not infer column lengths") {
```

Comment:
super nit: add a blank line between L263 and L264?

Reply:
OK done